mqtt.md

```diff
@@ -0,0 +1,94 @@
+This library allows for the execution of async ClearBlade messaging functions by providing an MQTT client that will work asynchronously. You can connect to the internal ClearBlade broker or an external broker. 
+
+## MQTT.Message(payload)
+
+The message represents an MQTT message.
+
+* @param {string} payload
+* @constructor
+* @property {string} payload
+* @property {number} qos
+* @property {boolean} retain
+* @property {boolean} duplicate
+
+Example
+
+~~~javascript
+var msg = new MQTT.Message("my message") 	
+~~~
+
+## MQTT.Client(options)
+
+The client represents an MQTT client connected to a broker.
+If no options are specified, the client connects to the ClearBlade broker.
+
+* @param {object} [options]
+* @param {string} options.address
+* @param {number} [options.port=1883] 
+* @param {string} [options.username] 
+* @param {string} [options.password]
+* @param {string} [options.client_id={randomUUID}]
+* @param {boolean} [options.use_tls=false]
+* @param {object} [options.tls_config]
+* @param {string} options.tls_config.client_cert
+* @param {string} options.tls_config.client_key
+* @param {string} options.tls_config.ca_cert
+* @constructor
+
+Example
+
+The code block below is to connect to an external clearblade broker. 
+
+~~~javascript
+var client = new MQTT.Client(options);
```

Comment:
Switch the order of options and client here so you define options before you use them.